lib/normalizer.c

```diff
@@ -1967,6 +1967,53 @@ grn_nfkc_normalize_unify_hiragana_voiced_sound_mark(
   return utf8_char;
 }
 
+static inline bool
+grn_nfkc_normalize_hiragana_ensure_voiced_sound_mark(
+  const unsigned char *utf8_char, unsigned char *voiced)
+{
+  if (utf8_char[0] == 0xe3 && utf8_char[1] == 0x81) {
+    if (0x8b <= utf8_char[2] && utf8_char[2] <= 0xa2) {
+      /* U+304B HIRAGANA LETTER KA ..
+       * U+3062 HIRAGANA LETTER DI */
+      voiced[0] = utf8_char[0];
+      voiced[1] = utf8_char[1];
+      voiced[2] = utf8_char[2];
+      if (utf8_char[2] & 0x1) {
+        voiced[2] += 1;
+      }
+      return true;
+    } else if (0xa4 <= utf8_char[2] && utf8_char[2] <= 0xa9) {
+      /* U+3064 HIRAGANA LETTER TU ..
+       * U+3069 HIRAGANA LETTER DO */
+      voiced[0] = utf8_char[0];
+      voiced[1] = utf8_char[1];
+      voiced[2] = utf8_char[2];
+      if (!(utf8_char[2] & 0x1)) {
+        voiced[2] += 1;
+      }
+      return true;
+    } else if (0xaf <= utf8_char[2] && utf8_char[2] <= 0xbd) {
+      /* U+306F HIRAGANA LETTER HA ..
+       * U+307D HIRAGANA LETTER PO */
+      unsigned char mod3 = (unsigned char)((utf8_char[2] - 0xaf) % 3);
+      if (mod3 == 0) {
+        /* Unvoiced -> add voiced mark */
+        voiced[0] = utf8_char[0];
+        voiced[1] = utf8_char[1];
+        voiced[2] = utf8_char[2] + 1;
+        return true;
+      } else if (mod3 == 1) {
+        /* mod3 == 1 (already voiced) -> no change */
```

Comment:
`mod3 == 1` is redundant.

```suggestion
        /* Already voiced -> no change */
```